tket/test/src/test_PauliGraph.cpp

```diff
@@ -693,6 +694,14 @@ SCENARIO("Test mutual diagonalisation of fully commuting sets") {
     test2.symbol_substitution(symbol_map);
     REQUIRE(test_statevector_comparison(test1, test2));
   }
+  GIVEN("A circuit with a classical bit") {
+    // https://github.com/CQCL/tket/issues/1578
+    Circuit c0(0, 1);
+    CircBox cbox(c0);
+    Circuit c(0, 1);
+    c.add_box<CircBox, Bit>(cbox, {Bit("c", 0)});
+    REQUIRE_NOTHROW(Transforms::special_UCC_synthesis().apply(c));
```

Comment:
I didn't know `GuidedPauliSimp` could handle classically controlled CircBox. Is this fix just to make sure that the method doesn't complain that there are classically controlled CircBox that it won't optimise anyway (in which case I think this test is fine), or is it also decomposing them (in which case maybe we need the CircBox to include something that gets optimised)?

Reply:
In this case there's no classical control, it's just a classical wire being passed through the `CircBox`. Previously this caused an error because the `Subcircuit` constructor was assuming all wires were quantum.